app/main.py

```diff
@@ -1,3 +1,66 @@
+
+
+class Knight:
+
+    def __init__(self, knight: dict) -> None:
+        self.protection = 0
+        self.name = knight["name"]
+        self.power = knight["power"]
+        self.hp = knight["hp"]
+        self.armour = knight["armour"]
+        self.weapon = knight["weapon"]
+        self.potion = knight["potion"]
+
+    def take_weapon(self) -> None:
+        self.power += self.weapon["power"]
+
+    def put_on_armor(self) -> None:
+        for armor in self.armour:
+            self.protection += armor["protection"]
+
+    def use_potion(self) -> None:
+        if self.potion:
+            potion_effect = self.potion["effect"]
+            for key, value in potion_effect.items():
+                if hasattr(self, key):  # Checking if the attribute exists
+                    current_value = getattr(self, key)
+                    setattr(self, key, current_value + value)
+
+    def get_ready_to_battle(self) -> None:
+        self.take_weapon()
+        self.put_on_armor()
+        self.use_potion()
+
+
+class BattleArena:
+
+    @staticmethod
+    def check_hp(warriors: list) -> None:
+        for warrior in warriors:
+            if warrior.hp <= 0:
+                print(f"{warrior.name} was defeated!")
+                warrior.hp = 0
+
+    @staticmethod
+    def fight(warriors: list) -> dict:
+        for warrior in warriors:
+            warrior.get_ready_to_battle()
+
+        warriors[0].hp -= warriors[2].power - warriors[0].protection
+        warriors[2].hp -= warriors[0].power - warriors[2].protection
+        warriors[1].hp -= warriors[3].power - warriors[1].protection
+        warriors[3].hp -= warriors[1].power - warriors[3].protection
+
+        BattleArena.check_hp(warriors)
+
+        return {
+            warriors[0].name: warriors[0].hp,
+            warriors[1].name: warriors[1].hp,
+            warriors[2].name: warriors[2].hp,
+            warriors[3].name: warriors[3].hp,
+        }
```

Comment:
You have already this code? Why do you define it again?

Reply:
I have error " ModuleNotFoundError: No module named 'knight'"